slack/resource_conversation.go

```diff
@@ -107,6 +114,17 @@ func resourceSlackConversationCreate(ctx context.Context, d *schema.ResourceData
 	isPrivate := d.Get("is_private").(bool)
 
 	channel, err := client.CreateConversationContext(ctx, name, isPrivate)
+	if err != nil && err.Error() == "name_taken" && d.Get("adopt_existing_channel").(bool) {
```

Comment:
if we get the 'name_taken' error, and the new 'adopt_exsting_channel' flag is true, then find the existing channel and use it instead of a newly created channel.